ringtail/storagemanager.py

```diff
@@ -838,68 +764,179 @@ def _generate_interaction_tuples(cls, interaction_dictionaries: list):
+                Pose_ID = (
+                    -1
+                )  # nonsensical table index to initialize row index if checking for duplicates
+                if self.duplicate_handling:
+                    Pose_ID = self.check_unique_results_row(result)
+
+                if Pose_ID != -1:  # row exists in table
+                    duplicates.append(Pose_ID)
+                    # row exist, evaluate if ignore or replace
+                    if self.duplicate_handling.upper() == "IGNORE":
+                        # do not add the new, duplicated row
+                        pass
+                    elif self.duplicate_handling.upper() == "REPLACE":
+                        # update the existing row with the new results
+                        # reformat sqlite query to update
+                        sql_replace = sql_insert.replace(
+                            "INSERT INTO Results", "UPDATE Results SET"
+                        )
+                        sql_replace = sql_replace.replace("VALUES", "=")
+                        sql_replace = sql_replace.replace(";", " WHERE Pose_ID = ?;")
+                        result.append(
+                            Pose_ID
+                        )  # add pose ID to the data being processed in sqlite statement
+                        cur.execute(sql_replace, result)
+
+                else:  # row does not exist
+                    duplicates.append(None)
+                    cur.execute(sql_insert, result)
+                    Pose_ID = cur.lastrowid
+                # create list of pose ids just processed
+                Pose_IDs.append(Pose_ID)
+
             self.conn.commit()
             cur.close()
 
+            return Pose_IDs, duplicates
+
         except sqlite3.OperationalError as e:
             raise DatabaseInsertionError("Error while inserting results.") from e
 
+    def check_unique_results_row(self, result_data):
+        """Checks if a pose ID is uniquely represented in the result table, based on the following [index in result_data] columns:
+        [0] LigName,
+        [1] receptor,
+        [20] about_x,
+        [21] about_y,
+        [22] about_z,
+        [23] trans_x,
+        [24] trans_y,
+        [25] trans_z,
+        [26] axisangle_x,
+        [27] axisangle_y,
+        [28] axisangle_z,
+        [29] axisangle_w,
+        [30] dihedrals,
+
+        #NOTE Please note that this method will only identify one duplicate in the table. If there are more than one duplicates, it will just deal with the earliest entry
+
+        Args:
+            result_data (list): data packet coming from the results processing
+
+        Raises:
+            DatabaseQueryError
+
+        Returns:
+            Pose_ID (int): returns the Pose_ID of the duplicate if found, returns -1 of no duplicate found
+
+        """
+        # create list of the data that is to be considered unique
+        unique_data_indeces = [0, 1, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30]
```

Comment:
should probably be indexes or indices, minor typo

Reply:
fixed